### PR TITLE
GEM-75 | Add external GET user groups

### PR DESCRIPTION
**Pull request by lawos98**

### CODE REVIEW

File: src/test/kotlin/pl/edu/agh/gem/external/dto/ExternalUserGroupsResponseTest.kt [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    should("return an empty list when there are no Groups") {
+        // given
+        val groups = listOf<Group>()
+
+        // when
+        val externalUserGroupsResponse = groups.toExternalUserGroupsResponse()
+
+        // then
+        externalUserGroupsResponse.groups.shouldBe(listOf())
```
Comment (pwgacek) [inline src/test/kotlin/pl/edu/agh/gem/external/dto/ExternalUserGroupsResponseTest.kt]:
```suggestion
        externalUserGroupsResponse.groups shouldBe listOf()
```

**Review by pwgacek — CHANGES_REQUESTED** [review]
What about GroupServiceTest?

File: src/main/kotlin/pl/edu/agh/gem/internal/service/GroupService.kt [hunk 2]
```diff
@@ -51,5 +55,8 @@ class GroupService(
 class MissingGroupException(joinCode: String) :
     RuntimeException("Failed to find group with joinCode:$joinCode")
 
+class UserWithoutGroup(userId: String) :
```
Comment (pwgacek) [inline src/main/kotlin/pl/edu/agh/gem/internal/service/GroupService.kt]:
```suggestion
class UserWithoutGroupException(userId: String) :
```

**Review by pwgacek — APPROVED** [review]

File: src/main/kotlin/pl/edu/agh/gem/internal/service/GroupService.kt (hunk 2 above)
Comment (lawos98) [reply src/main/kotlin/pl/edu/agh/gem/internal/service/GroupService.kt]: fixed

**Pull request merged**